frame/staking/src/pallet/mod.rs

```diff
@@ -644,19 +662,6 @@ pub mod pallet {
 			}
 			// `on_finalize` weight is tracked in `on_initialize`
 		}
-
-		fn integrity_test() {
-			sp_std::if_std! {
-				sp_io::TestExternalities::new_empty().execute_with(||
-					assert!(
-						T::SlashDeferDuration::get() < T::BondingDuration::get() || T::BondingDuration::get() == 0,
-						"As per documentation, slash defer duration ({}) should be less than bonding duration ({}).",
-						T::SlashDeferDuration::get(),
-						T::BondingDuration::get(),
-					)
-				);
-			}
-		}
```

Comment:
why is this removed? it was a good test no ?

Reply:
I am not sure why it was originally removed, but I think in general test like these are a slightly since these values are only bounded by `Get` they could be instantiated as `storage` or `static` (instead of `const`) and be manipulated at runtime. Instead these values should be changed to `const` if we want these values not to change at runtime either because of integrity test or they require re-weighing etc.

Reply:
I reverted it in 627bb160b13c2ee015a49ede2a3b6b779feafd93

Reply:
I see, I guess we can open another PR or issue to talk about this.